target/linux/realtek/base-files/etc/board.d/03_gpio_switches

```diff
@@ -0,0 +1,16 @@
+
+. /lib/functions/uci-defaults.sh
+
+board_config_update
+
+board=$(board_name)
+
+case "$board" in
+hpe,1920-8g-poe)
+	ucidef_add_gpio_switch "fan_ctrl" "Fan control" "456" "0"
```

Comment:
Sorry to be late to the party.

I don't see it in the commit message, but I assume this is an on/off control for the fan(s) in the switch? If it is, have you considered using a [`gpio-fan`](https://elixir.bootlin.com/linux/latest/source/Documentation/devicetree/bindings/hwmon/gpio-fan.txt) node instead?

If the device has three-wire fans, like the [Delta EFB0412HHD-R00](https://www.delta-fan.com/Download/Spec/EFB0412HHD-R00.pdf) in the 1920-24G-PoE+ (JG926A), there might also be a way to detect a stall (i.e. `alarm-gpios` property).

Reply:
Fan gpio changes fan speed. It could spin slower or faster.